pkg/apis/samples/v1alpha1/samplesource_validation.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+)
+
+// Validate validates SampleSource.
+func (s *SampleSource) Validate(ctx context.Context) *apis.FieldError {
+	var errs *apis.FieldError
+
+	//example: validation for "spec" field
+	errs = errs.Also(s.Spec.Validate(ctx).ViaField("spec"))
+
+	//errs is nil if everything is fine.
+	return errs
+}
+
+// Validate validates SampleSourceSpec.
+func (sspec *SampleSourceSpec) Validate(ctx context.Context) *apis.FieldError {
+	//Add code for validation webhook for SampleSourceSpec
+	var errs *apis.FieldError
+
```

Comment:
given that this is a sample source, specifically for beginners, it'd be great if we could actually show validation of our simple spec fields

Reply:
(ie, not just the sink)

Reply:
OK. I am going to add validation for the other fields.